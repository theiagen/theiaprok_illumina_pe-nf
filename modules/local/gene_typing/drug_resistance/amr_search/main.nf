process AMR_SEARCH {
    tag "$meta.id" 
    label "process_low"

    container "us-docker.pkg.dev/general-theiagen/theiagen/amrsearch:0.2.1"
    containerOptions '--entrypoint=""'

    input:
    tuple val(meta), path(assembly_fasta)
    val amr_search_database 

    output:
    tuple val(meta), path("*_amr_results.csv"), emit: amr_results_csv
    tuple val(meta), path("*_amr_results.pdf"), emit: amr_results_pdf
    tuple val(meta), path("*_paarsnp_results.json"), emit: amr_search_json_output
    path "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def prefix = task.ext.prefix ?: "${meta.id}"
    def database = amr_search_database ?: "485"
    """
    # Unzip the assembly if necessary
    REAL_FILE=\$(readlink -f "${assembly_fasta}")
    if [[ "\${REAL_FILE}" == *.gz ]] || [[ "\$(file -b "\${REAL_FILE}")" == *"gzip"* ]] || [[ "\$(file -b --mime-type "\${REAL_FILE}")" == "application/gzip" ]]; then
        zcat "\${REAL_FILE}" > assembly.fa
    else
        ln -sf "\${REAL_FILE}" assembly.fa
    fi
    # Run the tool
    java -jar /paarsnp/paarsnp.jar \
        -i assembly.fa \
        -s ${database}

    # Move the output file from the input directory to the working directory
    ln -sf assembly_paarsnp.jsn ./${prefix}_paarsnp_results.json

    # Script housed within the image; https://github.com/theiagen/theiagen_docker_builds/tree/awh-amrsearch-image/amrsearch/0.0.20
    python3 /scripts/parse_amr_json.py \
        ./${prefix}_paarsnp_results.json \
        ${prefix}

      
    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        amrsearch: \$(cat output_amr_version.txt)"
    END_VERSIONS
    """

}