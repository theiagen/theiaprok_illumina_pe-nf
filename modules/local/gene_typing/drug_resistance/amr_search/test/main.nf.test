nextflow_process {
    name "Test Process AMR_SEARCH"
    script "../main.nf"
    process "AMR_SEARCH"
    
    test("amr search test") {
        when {
            process {
                """
                input[0] = [
                    [ id:'test_sample' ],
                    file(params.pipelines_testdata_base_path + "genomics/prokaryotes/bacteroides_fragilis/genome/genome.fna.gz", checkIfExists: true)
                ]
                input[1] = "485" // amr_search_database
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                {  assert snapshot(
                    process.out.amr_results_csv,
                    process.out.amr_results_pdf,
                    process.out.amr_search_json_output,
                    process.out.versions
                    ).match() }
            )
        }
    }
    
//     test("should run with theiaprok workflow - stub") {
//         options "-stub"
//         when {
//             process {
//                 """
//                 input[0] = [
//                     [ id:'test_sample' ],
//                     file(params.pipelines_testdata_base_path + "genomics/prokaryotes/bacteroides_fragilis/genome/genome.fna.gz", checkIfExists: true)
//                 ]
//                 """
//             }
//         }
//         then {
//             assertAll(
//                 { assert process.success },
//                 { assert snapshot(process.out).match() }
//             )
//         }
//     }
}