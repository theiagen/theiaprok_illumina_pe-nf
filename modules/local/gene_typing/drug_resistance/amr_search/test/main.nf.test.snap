{
    "amr search test": {
        "content": [
            [
                [
                    {
                        "id": "test_sample"
                    },
                    "test_sample_amr_results.csv:md5,c2ddc073d73f04f6a84eeb640ea7ceb7"
                ]
            ],
            [
                [
                    {
                        "id": "test_sample"
                    },
                    "test_sample_amr_results.pdf:md5,43e4ff7992d7e476ce14c899b7c4b246"
                ]
            ],
            [
                [
                    {
                        "id": "test_sample"
                    },
                    "test_sample_paarsnp_results.json:md5,951a50fb36342194c7035bc6fff6f714"
                ]
            ],
            [
                "versions.yml:md5,fe8225ce22a17692b1c55e4e6165dde4"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.4"
        },
        "timestamp": "2025-06-30T15:55:09.519685265"
    }
}